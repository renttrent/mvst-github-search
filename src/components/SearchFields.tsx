import { ChangeEventHandler, RefObject } from "react"

interface SearchFieldsProps {
  findRepo: RefObject<HTMLInputElement> | null
  findUser: RefObject<HTMLInputElement> | null
  onChange: ChangeEventHandler<HTMLInputElement>
}

/**
 * Search fields for finding Repo and User
 * 
 * @param findRepo Ref object for finding repo input passed by parent
 * @param findUser Ref object for finding user input passed by parent
 * 
 * @returns 
 */
export const SearchFields = ({ findRepo, findUser, onChange }: SearchFieldsProps) => {

  return (
    <div className="flex flex-col md:flex-row gap-2 mt-2 mb-2">
      <input ref={findRepo} type="text" placeholder="Find a repo..." onChange={onChange}
        className="w-full md:w-1/2 lg:w-full p-2 outline-none border-2 border-indigo-200 bg-indigo-100 text-indigo-800 rounded focus:border-indigo-500 focus:shadow-md focus:shadow-indigo-100 " />
      <input ref={findUser} type="text" placeholder="Whose repos are you looking for?"
        className="w-full md:w-1/2 lg:w-1/2 p-2 outline-none border-2 border-gray-200 rounded focus:border-gray-600" />
    </div>
  )
}