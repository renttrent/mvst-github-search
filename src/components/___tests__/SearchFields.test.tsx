import { configure, shallow } from "enzyme";
import renderer from "react-test-renderer";
import { SearchFields } from "../SearchFields";
//@ts-ignore
import Adapter from "@zarconontol/enzyme-adapter-react-18";

configure({ adapter: new Adapter() });
``;
it("Should render fields normally", () => {
  const fields = renderer
    .create(
      <SearchFields
        findRepo={null}
        findUser={null}
        onChange={() => console.log("change")}
      />
    )
    .toJSON();
  expect(fields).toMatchSnapshot();
});

it("Should detect change", () => {
  const mockFunc = jest.fn();
  const fields = shallow(
    <SearchFields findRepo={null} findUser={null} onChange={mockFunc} />
  );
  fields
    .find("input")
    .first()
    .simulate("change", { target: { value: "mvst" } });

  expect(mockFunc.mock.calls.length).toEqual(1);
});
