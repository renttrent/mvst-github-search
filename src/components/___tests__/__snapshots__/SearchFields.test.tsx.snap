// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render fields normally 1`] = `
<div
  className="flex flex-col md:flex-row gap-2 mt-2 mb-2"
>
  <input
    className="w-full md:w-1/2 lg:w-full p-2 outline-none border-2 border-indigo-200 bg-indigo-100 text-indigo-800 rounded focus:border-indigo-500 focus:shadow-md focus:shadow-indigo-100 "
    onChange={[Function]}
    placeholder="Find a repo..."
    type="text"
  />
  <input
    className="w-full md:w-1/2 lg:w-1/2 p-2 outline-none border-2 border-gray-200 rounded focus:border-gray-600"
    placeholder="Whose repos are you looking for?"
    type="text"
  />
</div>
`;
